source/core/src/core/silver/domain/validations/enum_validations.py

```diff
@@ -5,29 +5,48 @@
     SubmittedTransactionsUnpackedColumnNames,
 )
 from core.contracts.process_manager.enums.metering_point_type import MeteringPointType
+from core.contracts.process_manager.enums.orchestration_type import OrchestrationType
+from core.contracts.process_manager.enums.quality import Quality
+from core.contracts.process_manager.enums.resolution import Resolution
+from core.contracts.process_manager.enums.unit import Unit
+from core.utility.columns_helper import column_is_not_null
 
 
 def validate_orchestration_type_enum() -> Column:
     """Quality check: QCST01-01."""
-    return F.lit(True)
+    valid_orchestration_types = [
+        name for name in OrchestrationType._member_names_ if name != OrchestrationType.OT_UNSPECIFIED.value
+    ]
+    return F.col(SubmittedTransactionsUnpackedColumnNames.orchestration_type).isin(valid_orchestration_types)
 
 
 def validate_quality_enum() -> Column:
     """Quality check: QCST01-02."""
-    return F.lit(True)
+    valid_qualities = [name for name in Quality._member_names_ if name != Quality.Q_UNSPECIFIED.value]
+
+    return column_is_not_null(SubmittedTransactionsUnpackedColumnNames.points).otherwise(
+        F.forall(
+            F.col(SubmittedTransactionsUnpackedColumnNames.Points.quality),
+            lambda x: x.isNotNull() & x.isin(valid_qualities),
+        )
+    )
```

Comment:
When look at the `column_is_not_null` method, it looks like we are checking if the column `isNull` and not `isNotNull`. 
```suggestion
    return F.when(is_null(SubmittedTransactionsUnpackedColumnNames.points), F.lit(False)).otherwise(
        F.forall(
            F.col(SubmittedTransactionsUnpackedColumnNames.Points.quality),
            lambda x: x.isNotNull() & x.isin(valid_qualities),
        )
    )
```

I think what confuses me is the `.otherwise` after the `column_is_not_null`.

To me, it looks like if the column is not null, then we return (?). Otherwise (if null), we check if all the points quality is valid.